CCP-171 Changes for wrapping concept while editing

mkarthik90 opened the pull request.

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java:
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
+        // Remove CCP concepts. 
+        for(ConceptEntryWrapper wrapper : foundConcepts) {
+            System.out.println(wrapper.getEntry().getId());
+            if (wrapper.getEntry().getId().startsWith(Constants.CONCEPT_PREFIX)) {
```
Isn't there a method to test concept ids for what type they are?

mkarthik90 replied:
@jdamerow  Yes there is a class called ConceptTypesService but that class has been added as part of CCP-163. May be i will create this PR once that is merged

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java:
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
+        
+        // Remove CCP concepts. 
+        for(ConceptEntryWrapper wrapper : foundConcepts) {
+            System.out.println(wrapper.getEntry().getId());
```
?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java:
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
+        }
+        
+        if (foundConcepts == null || foundConcepts.isEmpty()) {
+            return new ResponseEntity<Object>(HttpStatus.NOT_FOUND);
```
should just return empty results. nothing special here, just no results.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java:
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
         return new ResponseEntity<String>(buildJSON(synonymList, false, true), HttpStatus.OK);
     }
 
+    @RequestMapping(method = RequestMethod.GET, value = "conceptEdit/conceptSearch")
```
make that just .../search

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java:
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
     }
 
+    @RequestMapping(method = RequestMethod.GET, value = "conceptEdit/conceptSearch")
+    public @ResponseBody ResponseEntity<Object> searchConcept(@RequestParam("concept") String concept,
```
Doesn't this method do the same thing as the method that is called when creating a new concept? Can't you reuse, or combine these two?

mkarthik90 replied:
@jdamerow  Yes I was thinking the same. While creating the concept we are returning the model object but here we are returning the response entity. May be I can combine the logic of searching a concept. While create a new concept we also get the concept list and concept types details, Do we need the same details here too?

jdamerow replied:
Yes

mkarthik90 commented:
@jdamerow  Addressed review comments.

diging-jenkins commented:
Can one of the admins verify this patch?

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java:
```diff
@@ -258,6 +272,43 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
+        }
+        
+        // Remove CCP concepts. 
+        for(ConceptEntryWrapper wrapper : foundConcepts) {
```
If you search for only concepts in Wordnet list (line 292), do you need this here?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java:
```diff
@@ -258,6 +272,43 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
     }
 
+    @RequestMapping(method = RequestMethod.GET, value = "conceptEdit/search")
+    public @ResponseBody ResponseEntity<Object> searchConcept(@RequestParam("concept") String concept,
```
This code is still duplicating existing code, isn't it? Extract duplicate code and reuse it.

jdamerow replied:
Hm, ya, two lines are the same and the structure... I'll create a new ticket to refactor the wrapper creator. For now let's leave it the way it is.

mkarthik90 commented:
@jdamerow  Addressed review comments. 

I have not merged the logic of searching in ConceptWrapperAddController ConceptEditController. Only the code for fetching the concept entry looks similar. Please let me know if I need to still create a common method.

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented:
ok to test

jdamerow commented:
Let's make a little button next to each row that lets users add a concept to the wrapped concepts:
![image](https://cloud.githubusercontent.com/assets/8881141/20025457/c0d40976-a2ab-11e6-976f-b0cf7617232a.png)

jdamerow commented:
Also, the search shows regular concepts as well. Is the lucene search working correctly?

mkarthik90 commented:
Will merge to CCP-176
